src/plugins/search/search.js

```diff
@@ -1,35 +1,152 @@
-    var rowCount = instance.countRows();
-    var colCount = instance.countCols();
-    var queryResult = [];
-
-    if (!callback) {
-      callback = Search.global.getDefaultCallback();
-    }
+class Search extends BasePlugin {
+  constructor(hotInstance) {
+    super(hotInstance);
+    /**
+     * Callback function is responsible for marking the cells belonging to the search by setting property to it.
+     *
+     * @type {Function}
+     */
+    this.callback = DEFAULT_CALLBACK;
+    /**
+     * Query function is responsible for determining whether a query matches the value stored in a cell.
+     *
+     * @type {Function}
+     */
+    this.queryMethod = DEFAULT_QUERY_METHOD;
+    /**
+     *  Class added to each cell that belongs to the search.
+     *
+     * @type {String}
+     */
+    this.searchResultClass = DEFAULT_SEARCH_RESULT_CLASS;
+  }
+
+  /**
+   * Check if the plugin is enabled in the Handsontable settings.
+   *
+   * @returns {Boolean}
+   */
+  isEnabled() {
+    return this.hot.getSettings().search;
+  }
 
-    if (!queryMethod) {
-      queryMethod = Search.global.getDefaultQueryMethod();
+  /**
+   * Enable plugin for this Handsontable instance.
+   */
+  enablePlugin() {
+    if (this.enabled) {
+      return;
     }
 
-    for (var rowIndex = 0; rowIndex < rowCount; rowIndex++) {
-      for (var colIndex = 0; colIndex < colCount; colIndex++) {
-        var cellData = instance.getDataAtCell(rowIndex, colIndex);
-        var cellProperties = instance.getCellMeta(rowIndex, colIndex);
-        var cellCallback = cellProperties.search.callback || callback;
-        var cellQueryMethod = cellProperties.search.queryMethod || queryMethod;
-        var testResult = cellQueryMethod(queryStr, cellData);
+    const searchSettings = this.hot.getSettings().search;
+    this.checkPluginSettings(searchSettings);
+
+    this.addHook('beforeRenderer', (TD, row, col, prop, value, cellProperties) => this.onBeforeRenderer(TD, row, col, prop, value, cellProperties));
+
+    super.enablePlugin();
+  }
+
+  /**
+   * Disable plugin for this Handsontable instance.
+   */
+  disablePlugin() {
+    const beforeRendererCallback = (TD, row, col, prop, value, cellProperties) => this.onBeforeRenderer(TD, row, col, prop, value, cellProperties);
+
+    this.hot.addHook('beforeRenderer', beforeRendererCallback);
+    this.hot.addHookOnce('afterRender', () => {
+      this.hot.removeHook('beforeRenderer', beforeRendererCallback);
+    });
+
+    super.disablePlugin();
+  }
+
+  /**
+   * Updates the plugin to use the latest options you have specified.
+   */
+  updatePlugin() {
+    this.disablePlugin();
+    this.enablePlugin();
+
+    super.updatePlugin();
+  }
+
+  /**
+   * Query method - used inside search input listener.
+   *
+   * @param {String} queryStr Searched value.
+   * @param {Function} [callback=DEFAULT_CALLBACK] Callback function responsible for setting the cell's `isSearchResult` property.
```

Comment:
Please change from `[callback=DEFAULT_CALLBACK]` to `[callback]`. Apply the same rule below.

Reply:
Done.